primitives/io/src/lib.rs

```diff
@@ -87,16 +87,41 @@ pub enum EcdsaVerifyError {
 	BadSignature,
 }
 
+/// This is the initial kill storage result used
 /// The outcome of calling `storage_kill`. Returned value is the number of storage items
 /// removed from the trie from making the `storage_kill` call.
 #[derive(PassByCodec, Encode, Decode)]
-pub enum KillStorageResult {
+enum KillStorageResultPrev {
 	/// No key remains in the child trie.
 	AllRemoved(u32),
+
 	/// At least one key still resides in the child trie due to the supplied limit.
 	SomeRemaining(u32),
 }
 
+/// The outcome of calling `storage_kill`. Returned value is the number of storage items
+/// removed from the trie and also removed from overlay from making the `storage_kill` call.
+#[derive(PassByCodec, Encode, Decode)]
+pub enum KillStorageResult {
+	/// No key remains in the child trie.
+	AllRemoved {
+		/// number of keys remaining in backend
+		backend: u32,
+
+		/// number of keys remaining in overlay
+		overlay: u32
+	},
+
+	/// At least one key still resides in the child trie due to the supplied limit.
+	SomeRemaining {
+		/// number of keys in backend
+		backend: u32,
+
+		/// number of keys in overlay
+		overlay: u32
+	},
+}
```

Comment:
personally I would just make this a struct with 3 fields:
```rust
overlay: u32,
backend: u32,
all_removed: bool,
```

Reply:
@bkchr suggested the structs that way in one of our initial conversations.
Should i go ahead and implement your suggestion @thiolliere

Reply:
should i go ahead and implement your suggestion instead @thiolliere

Reply:
all in all it is fine anyway, I think we can keep it like it is currently

Reply:
Alright